doc/tutorials/error_analysis/error_analysis_part2.ipynb

```diff
@@ -0,0 +1,534 @@
+    "autocov = np.empty(1000)\n",
+    "\n",
+    "for j in range(1000):\n",
+    "    autocov[j] = np.dot(time_series_1_centered[:N_SAMPLES-j], time_series_1_centered[j:])\n",
+    "autocov /= N_SAMPLES\n",
+    "\n",
+    "plt.gca().axhline(0, color=\"gray\",linewidth=1)\n",
+    "plt.plot(autocov)\n",
+    "plt.xlabel(\"lag time\")\n",
+    "plt.ylabel(\"auto-covariance\")\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "compatible-speech",
+   "metadata": {},
+   "source": [
+    "We can see that the auto-covariance function starts at a high value and decreases quickly into a long noisy tail which fluctuates around zero. The high values at short lag times indicate that there are strong correlations at short time scales, as expected. However, even though the tail looks uninteresting, it can bear important information about the statistics of your data. Small systematic deviations from 0 in the tail can be a hint that long-term correlations exist in your system. On the other hand, if there is no sign of a systematic deviation from 0 in the tail, this usually means that the correlation is decaying well within the simulation time, and that the statistics are good enough to estimate an error. In the above example, the correlation quickly decays to zero. Despite the noise in the tail, the statistics seem very reasonable."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "lasting-deficit",
+   "metadata": {},
+   "source": [
+    "## Autocorrelation time\n",
+    "\n",
+    "Continuing our example, we can zoom into the first part of the auto-covariance function (using a log scale). We see that it indeed does have similarities with an exponential decay curve. In general, it isn't an exponential, but often can be approximated using one. If it matches reasonably well, the inverted prefactor in the exponential can be directly used as the *correlation time*, which is a measure for how many sampling intervals it takes for correlations to decay. Execute the following code cell for an illustration."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "protecting-graphic",
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "from scipy.optimize import curve_fit\n",
+    "\n",
+    "def exp_fnc(x,a,b):\n",
+    "    return a*np.exp(-x/b)\n",
+    "\n",
+    "N_MAX = 1000\n",
+    "j = np.linspace(0,N_MAX-1,N_MAX)\n",
+    "popt, pcov = curve_fit(exp_fnc, j, autocov[:N_MAX], p0=[15,10])\n",
+    "\n",
+    "plt.plot(j,autocov[:N_MAX])\n",
+    "plt.plot(j,exp_fnc(j,popt[0],popt[1]))\n",
+    "plt.xlim((1,N_MAX))\n",
+    "plt.xscale(\"log\")\n",
+    "plt.xlabel(\"lag time\")\n",
+    "plt.ylabel(\"auto-covariance\")\n",
+    "plt.show()\n",
+    "\n",
+    "print(\"Exponential autocorrelation time: {:.2f} sampling intervals\".format(popt[1]))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "behind-knife",
```

Comment:
I don't know what caused that, and I doubt it is necessary. If the notebook works as intended, including the hidden solutions, after the removal of these lines, they should be removed. Are you referring to `"id": "behind-knife",` ? Because I would guess that `"cell_type": "markdown",` is necessary to distinguish between code cells and markdown cells

Reply:
Yes, I just meant the ids. The are removed in af5396d446beaca8c5bbb96891aa4678d166ecc8.

Reply:
These fields are added in newer versions (4.5) of the notebooks. See [here](https://github.com/jupyterlab/jupyterlab/issues/9729). Since part 2 has been begun later than part 1, it is in notebook version 4.5, while the former is in 4.4 and doesn't contain any IDs. I'll revert the notebook to 4.4, so this won't occur. But likely we will have to deal with this in the future.

Reply:
Nice find! I'll try to keep that in mind the next time we generate the docker images with newer versions of Jupyter.